compass/landice/tests/ismip6_forcing/atmosphere/process_smb_racmo.py

```diff
@@ -0,0 +1,251 @@
+    """
+    A step for processing the regional RACMO surface mass balance data
+    """
+
+    def __init__(self, test_case, input_file=None):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.landice.tests.ismip6_forcing.atmosphere.Atmosphere
+            The test case this step belongs to
+
+        input_file : file name of ismip6 forcing data processed by this step
+        """
+        self.input_file = input_file
+        super().__init__(test_case=test_case, name='process_smb_racmo',
+                         ntasks=4, min_tasks=1)
+
+    def setup(self):
+        """
+        Set up this step of the test case
+        """
+        config = self.config
+        section = config["ismip6_ais"]
+        base_path_mali = section.get("base_path_mali")
+        mali_mesh_name = section.get("mali_mesh_name")
+        mali_mesh_file = section.get("mali_mesh_file")
+
+        section = config["ismip6_ais_atmosphere"]
+        process_smb_racmo = section.getboolean("process_smb_racmo")
+
+        self.add_input_file(filename=mali_mesh_file,
+                            target=os.path.join(base_path_mali,
+                                                mali_mesh_file))
+
+        if process_smb_racmo:
+            self.add_input_file(
+                target="RACMO2.3p2_ANT27_smb_yearly_1979_2018.nc",
+                database="RACMO2.3p2_ANT27_SMB_yearly_1979_2018")
+            output_file = f"{mali_mesh_name}_RACMO2.3p2_ANT27" \
+                          f"_smb_climatology_1995-2017.nc"
+
+            self.add_output_file(filename=output_file)
+        else:
+            print(f"\n'Warning: process_smb_racmo' is set to 'False'."
+                  f" This step will not run unless set 'True' in the"
+                  f" config file.\n")
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+
+        section = config["ismip6_ais"]
+        mali_mesh_name = section.get("mali_mesh_name")
+        mali_mesh_file = section.get("mali_mesh_file")
+        output_base_path = section.get("output_base_path")
+
+        section = config["ismip6_ais_atmosphere"]
+        method_remap = section.get("method_remap")
+
+        racmo_file_temp1 = "RACMO2.3p2_smb_climatology_1995_2017.nc"
+        racmo_file_temp2 = "RACMO2.3p2_smb_climatology_1995_2017_" \
+                           "correct_unit.nc"
+        output_file = f"{mali_mesh_name}_RACMO2.3p2_ANT27" \
+                      f"_smb_climatology_1995-2017.nc"
+
+        output_file_final = os.path.join(output_base_path, output_file)
+
+        if os.path.exists(output_file_final):
+            logger.info("Processed RACMO SMB data already exists in the "
+                        "output path {output_base_path}. "
+                        "Not processing the source data...")
+            return
+
+        input_file_list = self._files
```

Comment:
we got rid of `_files` in favor of using the compass database approach for RACMO.  As a result, you will need to get the input file from this step's `inputs` attribute, see:
https://github.com/MPAS-Dev/compass/blob/master/compass/step.py#L82-L85

It might be cleaner to just get the single input filename since that's all we will use:
```suggestion
        input_file = self.inputs[0]
```

Reply:
`self.inputs[0]` is a mali mesh file, but I've changed it to `self.inputs[1]` to correctly point to the original RACMO data.

Reply:
Okay, that's fine.